core/metacling/src/TCling.cxx

```diff
@@ -1536,6 +1536,10 @@ TCling::TCling(const char *name, const char *title, const char* const argv[], vo
                                                        llvmResourceDir, extensions,
                                                        interpLibHandle);
 
+   if (!fInterpreter->getCI()) { // Compiler instance could not be created. See https://its.cern.ch/jira/browse/ROOT-10239
+      std::cerr << "Exiting now since compiler instance is not available." << std::endl;
+      exit(EXIT_FAILURE);
```

Comment:
This is a library and we should not use `exit`. What about early return here?

Reply:
Yep, I tried that in the beginning but then it crashed later on at another time point. So I thought that for the time being it was better a clean exit than a straight crash.

Reply:
We should check in the caller for nullptr in that case.

Reply:
Isn't this in a constructor?  return early from a constructor will return an invalid object (not fully setup).  Do we have the proper checks already in place to verify validity?

Reply:
Follow up PR and comments: https://github.com/root-project/root/pull/16966/files#r1849138933